### PR TITLE
Declutter tags

### PR DESCRIPTION
**Pull request by napcs**
This PR reduces tags across the site, removing tags that look like keywords and reducing tags to a small set of tags related to Temporal capabilities.

This does not address keywords themselves, except for a few specific cases.

### CODE REVIEW

File: docs/cli/cmd-options.mdx [hunk 1]
```diff
@@ -113,113 +113,7 @@ keywords:
...
-  - workflow-execution
-  - workflow-id
-  - workflow-run
-  - workflow-state
-  - workflow-task
-  - workflow-task-failure
-  - workflow-type
-  - workflow-visibility
-  - x509-certificate
+  - Temporal CLI
```
Comment (jsundai) [inline docs/cli/cmd-options.mdx:116]: Looks great! There were a lot of this page - definitely made the page look a bit cluttered at the bottom.

File: docs/develop/dotnet/observability.mdx [hunk 2]
```diff
@@ -18,16 +18,11 @@ keywords:
...
-  - code-sample
-  - how-to
-  - logging
-  - search-attribute
-  - workflow
-  - observability
-  - sdk
-  - dotnet-sdk
-  - developer guide
+  - .Net SDK
```
Comment (jsundai) [inline docs/develop/dotnet/observability.mdx:21]: Do we want to have Metrics, Tracing, Logging as tags?

File: docs/develop/go/failure-detection.mdx [hunk 3]
```diff
@@ -4,12 +4,14 @@ title: Failure detection - Go SDK
...
 toc_max_heading_level: 4
 keywords:
-  - failure-detection
+  - failure detection
 tags:
-  - failure-detection
-  - sdk
-  - go-sdk
-  - developer guide
+  - Activities
```
Comment (jsundai) [inline docs/develop/go/failure-detection.mdx:9]: Would the tags here be Timeout, Retry, Heartbeat? Or does it stay more general?

File: docs/develop/dotnet/asynchronous-activity.mdx [hunk 4]
```diff
@@ -22,11 +22,9 @@ keywords:
...
   - getasyncactivityhandle
 tags:
-  - asynchronous-activity
-  - overview
-  - sdk
-  - dotnet-sdk
-  - developer guide
+  - Activities
+  - .Net SDK
+  - Temporal SDKs
```
Comment (dandavison) [inline docs/develop/dotnet/asynchronous-activity.mdx:27]: Is "Temporal SDKs" a useful tag for users?

Comment (fairlydurable) [reply docs/develop/dotnet/asynchronous-activity.mdx:27]: It would certainly get a lot of results, I'd imagine

Comment (napcs) [reply docs/develop/dotnet/asynchronous-activity.mdx:27]: I am leaning toward dropping it but wasn't sure.

**Review by fairlydurable — APPROVED** [review]
Unblocking this ticket.

**Review by jsundai — APPROVED** [review]

**Review by angelazhou32 — APPROVED** [review]

File: docs/develop/dotnet/asynchronous-activity.mdx (hunk 4 above)
Comment (napcs) [reply docs/develop/dotnet/asynchronous-activity.mdx:27]: Leaving for now and will measure.

**Pull request merged**